### ISSUE TITLE
regression in jsdoc/require-template in 62.0.0

### ISSUE THREAD

**Issue by alexander-akait** [gajus/eslint-plugin-jsdoc#1628]
Code:

```
/**
 * @template T
 * @typedef {T extends Record<string, Record<string, infer F>> ? F : never} ExtractFunction
 */
```

## Expected behavior

No errors

## Actual behavior

```
Missing @template F  jsdoc/require-template
```

## ESLint Config

```js
{
  rules: {
    "jsdoc/require-template": "error",
  }
}
```

## ESLint sample

```js
/**
 * @template T
 * @typedef {T extends Record<string, Record<string, infer F>> ? F : never} ExtractFunction
 */
```

## Environment

- Node version: v24.11.1
- ESLint version v9.39.2
- `eslint-plugin-jsdoc` version: v62.0.0

**Comment by brettz9** [comment]
I'm not convinced this was a regression since we didn't previously have any infer support, unless you began to see an issue because `infer` clauses began to be supported by our type parser. In any case, this should be fixed for at least basic cases now.

**Comment by alexander-akait** [comment]
@brettz9 Thanks for the quick fix
